x-pack/plugins/security_solution/common/detection_engine/schemas/common/schemas.ts

```diff
@@ -65,6 +65,18 @@ export type EventCategoryOverride = t.TypeOf<typeof event_category_override>;
 export const eventCategoryOverrideOrUndefined = t.union([event_category_override, t.undefined]);
 export type EventCategoryOverrideOrUndefined = t.TypeOf<typeof eventCategoryOverrideOrUndefined>;
 
+export const tiebreaker_field = t.string;
+export type TiebreakerField = t.TypeOf<typeof tiebreaker_field>;
+
+export const tiebreakerFieldOrUndefined = t.union([tiebreaker_field, t.undefined]);
+export type TiebreakerFieldOrUndefined = t.TypeOf<typeof tiebreakerFieldOrUndefined>;
+
+export const timestamp_field = t.string;
+export type TimestampField = t.TypeOf<typeof timestamp_field>;
+
+export const timestampFieldOrUndefined = t.union([timestamp_field, t.undefined]);
+export type TimestampFieldOrUndefined = t.TypeOf<typeof timestampFieldOrUndefined>;
```

Comment:
Looks like we may phasing out the `OrUndefined` pattern presently in place -- more details on that over here in this comment https://github.com/elastic/kibana/pull/132409#discussion_r878711020. No change needed, but just a heads up! 🙂

Reply:
Yup, that sounds good to me. I was also, not quite sure about the reason of having extra `OrUndefined` type :-)

Reply:
Great that we all agree!